File: .claude/skills/agent-skill-kit-commons/packages/mjs/bin/dump-score.mjs
#!/usr/bin/env node
// dump-score <descriptions.json> — emits canonical JSON of scoreDescription per case.
import { readFileSync } from "node:fs";
import { scoreDescription } from "../src/lint/scorer.mjs";

const path = process.argv[2];
if (!path) {
  process.stderr.write("usage: dump-score <descriptions.json>\n");
  process.exit(64);
}
const cases = JSON.parse(readFileSync(path, "utf8"));
const out = {};
for (const c of cases) {
  const { score, breakdown } = scoreDescription(c.description, c.body);
  out[c.id] = { score, breakdown };
}
process.stdout.write(JSON.stringify(sortKeys(out), null, 2) + "\n");

function sortKeys(v) {
  if (Array.isArray(v)) return v.map(sortKeys);
  if (v && typeof v === "object") {
    const o = {};
    for (const k of Object.keys(v).sort()) o[k] = sortKeys(v[k]);
    return o;
  }
  return v;
}

File: .claude/skills/agent-skill-kit-commons/packages/py/src/skills_ref/scripts/dump_score.py
from __future__ import annotations

import json
import sys

from skills_ref.lint.scorer import score_description


def _sort(v):
    if isinstance(v, list):
        return [_sort(x) for x in v]
    if isinstance(v, dict):
        return {k: _sort(v[k]) for k in sorted(v.keys())}
    return v


def main() -> int:
    if len(sys.argv) < 2:
        sys.stderr.write("usage: dump-score <descriptions.json>\n")
        return 64
    cases = json.loads(open(sys.argv[1], encoding="utf-8").read())
    out: dict = {}
    for c in cases:
        r = score_description(c["description"], c["body"])
        out[c["id"]] = {"score": r["score"], "breakdown": r["breakdown"]}
    sys.stdout.write(
        json.dumps(_sort(out), indent=2, separators=(",", ": ")) + "\n"
    )
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
